File: CS/20_Print/SetPrintSettingsByPrinterSettings.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using System.Drawing.Printing;

namespace SetPrintSettingsByPrinterSettings
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PowerPoint document.
            Presentation presentation = new Presentation();

            //Load the file from disk.
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Ppt_6.pptx");

            //Use PrinterSettings object to print presentation slides.
            PrinterSettings ps = new PrinterSettings();
            ps.PrintRange = PrintRange.AllPages;
            ps.PrintToFile = true;
            String result = "Result-SetPrintSettingsByPrinterSettingsObject.xps";
            ps.PrintFileName = (result);

            //Print the slide with frame.
            presentation.SlideFrameForPrint = true;

            //Print the slide with Grayscale.
            presentation.GrayLevelForPrint = true;

            //Print 4 slides horizontal.
            presentation.SlideCountPerPageForPrint = PageSlideCount.Four;
            presentation.OrderForPrint = Order.Horizontal;

            //Only select some slides to print.
            //presentation.SelectSlidesForPrint("1", "3");

            //Print the document.
            presentation.Print(ps);

            //Launch the file.
            PptDocumentViewer(result);
        }

        private void PptDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/20_Print/SetPrintSettingsByPrinterSettings.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports System.Drawing.Printing

Namespace SetPrintSettingsByPrinterSettings
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PowerPoint document.
			Dim presentation As New Presentation()

			'Load the file from disk.
			presentation.LoadFromFile("..\..\..\..\..\..\Data\Template_Ppt_6.pptx")

			'Use PrinterSettings object to print presentation slides.
			Dim ps As New PrinterSettings()
			ps.PrintRange = PrintRange.AllPages
			ps.PrintToFile = True
			Dim result As String = "Result-SetPrintSettingsByPrinterSettingsObject.xps"
			ps.PrintFileName = (result)

			'Print the slide with frame.
			presentation.SlideFrameForPrint = True

			'Print the slide with Grayscale.
			presentation.GrayLevelForPrint = True

			'Print 4 slides horizontal.
			presentation.SlideCountPerPageForPrint = PageSlideCount.Four
			presentation.OrderForPrint = Order.Horizontal

			'Only select some slides to print.
			'presentation.SelectSlidesForPrint("1", "3");

			'Print the document.
			presentation.Print(ps)

			'Launch the file.
			PptDocumentViewer(result)
		End Sub

		Private Sub PptDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace